### ISSUE TITLE
update JavadocTokenTypes.java to new format of AST print

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#14631]
Same as https://github.com/checkstyle/checkstyle/issues/14516 but for javadoc 

https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java

https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.html

https://checkstyle.org/cmdline.html , see -j or -J .

```
SteLeo@DESKTOP-11C7U5R MINGW64 ~/exception_literal test
$ cat Test.java
* @exception SQLException if query is not correct
SteLeo@DESKTOP-11C7U5R MINGW64 ~/exception_literal test
$ java -jar checkstyle-10.16.0-all.jar -j Test.java | sed "s/\[[0-9]\+:[0-9]\+\]//g"
JAVADOC -> JAVADOC
|--LEADING_ASTERISK -> *
|--WS ->
|--JAVADOC_TAG -> JAVADOC_TAG
|   |--EXCEPTION_LITERAL -> @exception
|   |--WS ->
|   |--CLASS_NAME -> SQLException
|   |--WS ->
|   `--DESCRIPTION -> DESCRIPTION
|       `--TEXT -> if query is not correct
`--EOF -> <EOF>
```
Good example on how update should looks like is at https://github.com/checkstyle/checkstyle/pull/14862

To generate AST, use cmd :  `java -jar checkstyle-10.13.0-all.jar -J Test.java | sed "s/\[[0-9]\+:[0-9]\+\]//g"`
The sed command removes the line:column numbers for our documentation.

Each token update should be in its own Pull Request.
Share whole  cli output for javadoc chunk in PR description.

~ATTENTION: do not forget to escape `>` to `&gt`. And also escape `<` if appeared.~

**Watch videos on how to do new PR and do exactly this update , link to youtube video is at this page** https://checkstyle.org/beginning_development.html#Starting_Development 

Good example of update: https://github.com/checkstyle/checkstyle/pull/14632 please provide all details as provided in description of this PR
One more good example of code changes: https://github.com/checkstyle/checkstyle/pull/16086/files

Documentation should be improved for:
- [x] AREA_HTML_TAG_NAME  
- [x] AREA_TAG  
- [x] ARGUMENT  
- [ ] ATTR_VALUE  
- [x] ATTRIBUTE 
- [x] AUTHOR_LITERAL  
- [x] BASE_HTML_TAG_NAME  
- [ ] BASE_TAG  
- [ ] BASEFONT_HTML_TAG_NAME  
- [ ] BASEFONT_TAG  
- [ ] BODY  
- [x] BODY_HTML_TAG_NAME  
- [x] BODY_TAG_END  
- [x] BODY_TAG_START  
- [x] BR_HTML_TAG_NAME  
- [x] BR_TAG  
- [ ] CHAR  
- [x] CLASS_NAME  
- [ ] CODE_LITERAL  
- [ ] COL_HTML_TAG_NAME 
- [ ] COL_TAG 
- [x] COLGROUP  
- [x] COLGROUP_HTML_TAG_NAME  
- [ ] COLGROUP_TAG_END  
- [ ] COLGROUP_TAG_START  
- [x] COMMA 
- [x] CUSTOM_NAME 
- [x] DD  
- [x] DD_HTML_TAG_NAME  
- [ ] DD_TAG_END  
- [ ] DD_TAG_START  
- [x] DEPRECATED_LITERAL  
- [ ] DESCRIPTION 
- [x] DOC_ROOT_LITERAL  
- [x] DT  
- [x] DT_HTML_TAG_NAME  
- [ ] DT_TAG_END  
- [ ] DT_TAG_START  
- [ ] EMBED_HTML_TAG_NAME 
- [x] EMBED_TAG 
- [ ] HTML 
- [x] EMPTY_TAG 
- [ ] END 
- [ ] EOF 
- [ ] EQUALS  
- [x] EXCEPTION_LITERAL 
- [x] FIELD_NAME  
- [x] FIELD_TYPE  
- [x] FRAME_HTML_TAG_NAME 
- [x] FRAME_TAG 
- [x] HASH  
- [ ] HEAD  
- [x] HEAD_HTML_TAG_NAME  
- [ ] HEAD_TAG_END  
- [ ] HEAD_TAG_START  
- [ ] HR_HTML_TAG_NAME  
- [x] HR_TAG  
- [ ] HTML  
- [ ] HTML_COMMENT  
- [ ] HTML_COMMENT_END  
- [ ] HTML 
- [ ] HTML_COMMENT_START  
- [ ] HTML 
- [x] HTML_ELEMENT  
- [ ] HTML_ELEMENT_END  
- [ ] HTML_ELEMENT_START  
- [ ] HTML_HTML_TAG_NAME  
- [ ] HTML_TAG  
- [ ] HTML_TAG_END  
- [ ] HTML_TAG_NAME 
- [ ] HTML_TAG_START  
- [ ] IMG_HTML_TAG_NAME 
- [x] IMG_TAG 
- [x] INHERIT_DOC_LITERAL 
- [x] INPUT_HTML_TAG_NAME 
- [x] INPUT_TAG 
- [ ] ISINDEX_HTML_TAG_NAME 
- [ ] ISINDEX_TAG 
- [x] JAVADOC 
- [x] JAVADOC_INLINE_TAG  
- [x] JAVADOC_INLINE_TAG_END  
- [x] JAVADOC_INLINE_TAG_START  
- [x] JAVADOC_TAG 
- [ ] KEYGEN_HTML_TAG_NAME  
- [ ] KEYGEN_TAG  
- [ ] HTML 
- [ ] LEADING_ASTERISK  
- [ ] LEFT_BRACE  
- [x] LI  
- [ ] LI_HTML_TAG_NAME  
- [ ] LI_TAG_END  
- [ ] LI_TAG_START  
- [ ] LINK_HTML_TAG_NAME  
- [ ] LINK_LITERAL  
- [ ] LINK_TAG  
- [x] LINKPLAIN_LITERAL 
- [x] LITERAL_EXCLUDE 
- [x] LITERAL_INCLUDE 
- [x] LITERAL_LITERAL 
- [x] MEMBER  
- [ ] A 
- [x] META_HTML_TAG_NAME  
- [x] META_TAG  
- [ ] NEWLINE 
- [ ] OPTGROUP  
- [ ] OPTGROUP_HTML_TAG_NAME  
- [ ] OPTGROUP_TAG_END  
- [ ] OPTGROUP_TAG_START  
- [ ] OPTION  
- [ ] OPTION_HTML_TAG_NAME  
- [ ] OPTION_TAG_END  
- [x] OPTION_TAG_START  
- [x] P_HTML_TAG_NAME 
- [ ] P_TAG_END 
- [ ] P_TAG_START 
- [x] PACKAGE_CLASS 
- [x] PACKAGE_CLASS 
- [x] PARAGRAPH 
- [ ] PARAM_HTML_TAG_NAME 
- [x] PARAM_LITERAL 
- [ ] PARAM_TAG 
- [x] PARAMETER_NAME  
- [x] PARAMETERS  
- [ ] RB  
- [ ] RB_HTML_TAG_NAME  
- [ ] RB_TAG_END  
- [ ] RB_TAG_START  
- [ ] REFERENCE 
- [x] RETURN_LITERAL  
- [x] RIGHT_BRACE 
- [ ] RP  
- [x] RP_HTML_TAG_NAME  
- [ ] RP_TAG_END  
- [ ] RP_TAG_START  
- [ ] RT  
- [ ] RT_HTML_TAG_NAME  
- [ ] RT_TAG_END  
- [ ] RT_TAG_START  
- [ ] RTC 
- [ ] RTC_HTML_TAG_NAME 
- [ ] RTC_TAG_END 
- [ ] RTC_TAG_START 
- [x] SEE_LITERAL 
- [x] SERIAL_DATA_LITERAL 
- [x] SERIAL_FIELD_LITERAL  
- [x] SERIAL_LITERAL  
- [x] SINCE_LITERAL 
- [ ] SINGLETON_ELEMENT 
- [x] SLASH 
- [ ] SLASH_END 
- [ ] SOURCE_HTML_TAG_NAME  
- [ ] SOURCE_TAG  
- [ ] HTML 
- [ ] START 
- [x] STRING  
- [ ] TBODY 
- [x] TBODY_HTML_TAG_NAME 
- [ ] TBODY_TAG_END 
- [ ] TBODY_TAG_START 
- [ ] TD  
- [x] TD_HTML_TAG_NAME  
- [ ] TD_TAG_END  
- [ ] TD_TAG_START  
- [ ] TEXT  
- [ ] CHAR 
- [ ] TFOOT 
- [ ] TFOOT_HTML_TAG_NAME 
- [ ] TFOOT_TAG_END 
- [ ] TFOOT_TAG_START 
- [ ] TH  
- [x] TH_HTML_TAG_NAME  
- [ ] TH_TAG_END  
- [ ] TH_TAG_START  
- [ ] THEAD 
- [x] THEAD_HTML_TAG_NAME 
- [ ] THEAD_TAG_END 
- [ ] THEAD_TAG_START 
- [x] THROWS_LITERAL  
- [x] TR  
- [x] TR_HTML_TAG_NAME  
- [ ] TR_TAG_END  
- [ ] TR_TAG_START  
- [ ] TRACK_HTML_TAG_NAME 
- [x] TRACK_TAG 
- [ ] HTML 
- [x] VALUE_LITERAL 
- [x] VERSION_LITERAL 
- [ ] WBR_HTML_TAG_NAME 
- [x] WBR_TAG 
- [ ] HTML 
- [ ] WS

**Comment by MANISH-K-07** [comment]
To generate AST, use cmd : ` java -jar checkstyle-10.13.0-all.jar -J Test.java | sed "s/\[[0-9]\+:[0-9]\+\]//g"`
The sed command removes the line:column numbers for our documentation.

**Comment by nrmancuso** [comment]
> @romani @nrmancuso If we don't create a UT to enforce this and ensure ANTLR updates are updated appropriately in this file and regular TokenTypes, then things like [#14615 (comment)](https://github.com/checkstyle/checkstyle/pull/14615#discussion_r1528684583) will break everything trying to be done in this issue.

I don't disagree that this would be nice to have, but I think this issue concerns the actual format that the AST is printed with.

**Comment by Prathamesh-007** [comment]
Hello. Is this issue assigned? If not, I would like to work on it.
I just have some doubts as this is my first contribution.

**Comment by romani** [comment]
Yes such update is expected .

No assignments, just make a comment "I am on xxxxx" and send PR

**Comment by Prathamesh-007** [comment]
I am on SERIAL_FIELD_LITERAL

**Comment by Kushagra-Sedwal** [comment]
is this issue open?

**Comment by romani** [comment]
Yes, there are bunch of examples of AST to update , just choose any that are left and send PR

**Comment by Kushagra-Sedwal** [comment]
@romani Please review the pr sent

**Comment by mernaahmed2020** [comment]
hello @romani i want to work on this issu, canyou assign me please.thanks in advance

**Comment by romani** [comment]
No assignments.
Just post here "i am on xxxxx" and start working on update.

**Comment by Rohan-here** [comment]
is this open ? i see pr was merged

**Comment by romani** [comment]
This issue designed to have numerous PRs. Please send one more update to examples that still in old format.

Each PR, has a single update of javadoc comment, but this class has numerous such javadoc comments.

**Comment by romani** [comment]
@Rohan-here, please help to update 
https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.html#CUSTOM_NAME

**Comment by romani** [comment]
I placed in description list of tokens that do not have javadoc or still have not-expected ast format that should be changed to expected. PRs are welcome, there are still a lot of work to be done.

**Comment by Akshat-Dimri** [comment]
is the issue still open ?, i'd like to try creating a pr for this

**Comment by romani** [comment]
PRs are welcome, there are still a lot of work to be done.
There is no assignments, as there suppose to be hundreds of PR to be opened, please choose token and send PR

**Comment by gupta-devang** [comment]
I will work on LITERAL_EXCLUDE

**Comment by praveenyadahalli** [comment]
Hi @romani, i gave fix for it yestarday but still it's in open status.

**Comment by romani** [comment]
I see exception_literal as checked

**Comment by praveenyadahalli** [comment]
> I see exception_literal as checked

Sorry if i am wrong, exception_literal is checked in list but when i see the problem description, it's still saying about problem exception_literal.

![Image](https://github.com/user-attachments/assets/6df28bdb-9c14-4eb7-8d9c-dab15e61cb84)

**Comment by romani** [comment]
It is just example, it will stay like this.

**Comment by praveenyadahalli** [comment]
> It is just example, it will stay like this.

Ok got it @romani

**Comment by Pranjali784** [comment]
Is this issue still open, can i contribute to it?

**Comment by romani** [comment]
Yes, there a lot of tokens not done, see check boxes in description.

**Comment by Pranjali784** [comment]
Ok thankyou for your reply , i'll look into the issue and will try to
contribute.

**Comment by vishakha-mali** [comment]
Hey , allow me also to try out the issue.

**Comment by romani** [comment]
Just choose non checked token in list and send PR

**Comment by vishakha-mali** [comment]
Can u check the code for slash token before applying for PR.Am I going on right track?

<img width="592" alt="Image" src="https://github.com/user-attachments/assets/56bcc613-81ec-4452-85e1-80d96caeb1c3" />

**Comment by romani** [comment]
Good, please send PR, can you check if escaping is actually required, if work without escaping, better to not escape

**Comment by Prasad2604** [comment]
Is this issue resolved completely? Or can I contribute to it?

**Comment by Anmol202005** [comment]
> Is this issue resolved completely? Or can I contribute to it?

@Prasad2604 Please read the issue description carefully, there is plenty of work to do for this one.

**Comment by Brijeshthummar02** [comment]
@romani can you mention one or two of it for which i should update JavadocTokenTypes.java to new format of AST print as there are almost 500+ chat so i am confused a bit.

**Comment by YuktiNandwana** [comment]
Hi @romani 

![Image](https://github.com/user-attachments/assets/11966e78-58f7-4c04-96e6-11288c25a037)

, I’m Yukti Nandwana. After setting up the Checkstyle project according to the video, I found that several errors are showing up when I inspect the code. Please guide me on what to do next.

**Comment by romani** [comment]
You can ignore inspection violations at the beginning, it looks like you misconfigured your local IDEA

**Comment by mobakir24** [comment]
I am on it

**Comment by khushikushwah** [comment]
After commit the issue what should I do.

![Image](https://github.com/user-attachments/assets/73403c70-fd3f-43e1-baa6-408569fcb750)

**Comment by mobakir24** [comment]
You should do PR

**Comment by baziz-meriem** [comment]
Hello , Is this issue still open?

**Comment by Anujanthwal-dotcom** [comment]
I am on THEAD_HTML_TAG_NAME.

**Comment by tomrist9** [comment]
I am on it

**Comment by pree-T** [comment]
Is this issue, still available to work on?

**Comment by Brijeshthummar02** [comment]
> Is this issue, still available to work on?

@pree-T you can select any of it, which is pending or needs to be updated.
very much, still there are multiple JavadocTokenTypes which need to be in new format of AST.

**Comment by VinishaWagh** [comment]
Hello! I would like to contribute to this issue..

**Comment by Brijeshthummar02** [comment]
> Hello! I would like to contribute to this issue..

Sure, checkout the pending ones - and pick any of it.

**Comment by jha9262** [comment]
Hi, I see multiple PRs opened for this. Are you still looking for new contributors, or should I try another issue?

**Comment by romani** [comment]
Please contribute

**Comment by romani** [comment]
We migrated to new TokensType class, so this issue is closed.

@mahfouz72 , will open new issue

**Comment by romani** [comment]
Activity is moved to https://github.com/checkstyle/checkstyle/issues/17882
Same type of updates are required there
